Sources/TSCBasic/FileSystem.swift

```diff
@@ -370,8 +370,9 @@ private struct LocalFileSystem: FileSystem {
     }
 
     func isSymlink(_ path: AbsolutePath) -> Bool {
-        let attrs = try? FileManager.default.attributesOfItem(atPath: path.pathString)
-        return attrs?[.type] as? FileAttributeType == .typeSymbolicLink
+        let url = NSURL(fileURLWithPath: path.pathString)
+        let result = try? url.resourceValues(forKeys: [.isSymbolicLinkKey])
+        return (result?[.isSymbolicLinkKey] as? Bool) == true
```

Comment:
should we continue to use `attributesOfItem` on non-darwin platforms? or this is cross-platform?

Reply:
It works cross-platform, there's just no performance benefit on non-Darwin.

Reply:
I'd prefer this code to have an in-place code comment that gives more background on why specifically `NSURL` is used here, and it could also link to the relevant PR/issue. Then for someone reading this code years later there would be no temptation to revert this change to something else without knowing the context.

Reply:
I really don't have any more background here tbh, I was using `NSURL` simply because the code didn't work with `URL`. I have long given up to understand why we decided to make Foundation hard to use in Swift.

Reply:
Right, I mean clarifying in code comments no the difference between `URL` and `NSURL`, but the fact that `NSURL.resourceValues` is faster than `FileManager.default.attributesOfItem` on certain platforms, so that this code is not reverted to the latter

Reply:
Oh I see, yah, I can add that.